### ISSUE TITLE
There is evidence that 'more distant' nodes have a voting advantage

### ISSUE THREAD

**Issue by bji** [solana-labs/solana#17567]
#### Problem

Through observation it has been noted that some nodes which have a typically higher 'average vote distance' also seem to have a better voting rate.  There appears to be a correlation between latency to the node from 'the epicenter', and successful voting percentage.  One theory is that seeing blocks later gives a validator a little bit of an advantage in making decisions on which fork to vote on -- perhaps being more likely to see both sides of a fork and being able to choose the more likely to succeed block, versus a validator who may see forks with less latency and therefore more quickly vote on 'the wrong' fork?

Some data:

Shinobi Systems (my node) has had the top or nearly the top voting performance every epoch that I have tracked it.  The only outlier was epoch 184, where my validator for some reason started out the epoch with a poor voting record and never recovered.  That being said, it still achieved better than average voting performance even that epoch.  Currently in epoch 185 it has landed over 5% more votes than the average.  The second place vote count is 2.2% behind it.  There is something significant going on, and it would good to understand it and to decide whether any adjustments to how the network operates is warranted.

My ping times to large European data centers are as follows:
Ping to P2P.org (199610-DE-Frankfurt): 143 ms
Ping to AG (199610-DE-Frankfurt): 143 ms
Ping to Stakin (24940-FI-Helsinki): 159 ms
ping to Forbole (24940-FI-Helsinki): 159 ms

This has been noted as being about 60 ms slower than nodes on the US east coast.

On validators.app, I can see that my node typically has worse than average 'vote distance'.

GenesysGo currently has the second most votes, and it also has worse than average 'vote distance'.

Additionally, I wonder if there is any hardware configuration aspect to this?  There are other nodes on the US west coast that don't have worse than average vote distance - for example, vote account B6nDYYLc2iwYqY3zdmavMmU9GjUL2hf79MkufviM2bXv in Los Angeles - and also don't have a correspondingly high vote rate.  'troo' and 'zorian' are in Las Vegas and similarly have good 'vote distance' and average 'vote rate'.

I am using a single RTX 2080ti GPU.  Is it possible that the code paths associated with GPU usage cause *worse vote distance* somehow, like they interfere with voting in some way, making it take *longer* to verify a block?  Or maybe it's the fact that the GPU can allow verification of more blocks in parallel, and the algorithm for choosing which block to vote for then naturally has more blocks to choose from?

This issue exists just as a place to track possible hypothesis for this situation and gather data.

**Comment by bji** [comment]
Here are logs covering two epochs plus a partial epoch:

Epoch 184 begin with block 79488000 and appears to first be seen around line 16116599 of validator-mainnet-beta.log.7.  This epoch saw my validator having average vote performance.  The epoch was exceptionally long (3 days 4 hours) indicating some kind of unusual problems with the network.  Skip rate was very high.

Epoch 185 began with block 79920000 which appears to first be seen around line 155785 of validator-mainnet-beta.log.3.  This epoch saw my validator having well above average vote performance -- the best performance of any validator in the epoch by a considerable margin.  The epoch was not as long (3 days 1 hour) -- still a high skip rate, but not quite as high.

Epoch 186 began with block 80352000 which appears to first be seen around line 1429010 of validator-mainnet-beta.log.  This epoch is in progress but again my validator is having very average vote rates.  The epoch is about as long and problematic as 184 was - estimated at 3 days 4 hours again.  Not sure if that is a factor.

https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.gz
https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.1.gz
https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.2.gz
https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.3.gz
https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.4.gz
https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.5.gz
https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.6.gz
https://solana-labs-issue-17567.s3-us-west-2.amazonaws.com/validator-mainnet-beta.log.7.gz

**Comment by bji** [comment]
Another small bit of data: in the current epoch, where my vote performance is average, my node appears to either be very close to 'the tip' of vote distance, or much further away (i.e. it stays close but occasionally drops far behind).  In other epochs where I performed much better, I believe that I was consistently an average of 3 - 4 blocks behind.

Another validator - GenesysGo - appears to be currently in the 'consistently 3 - 4 epochs behind' situation and is now the leading voter.

There appears to be a correlation between being consistently slightly behind, and having better voting performance.

Finally: I upgraded to 1.6.10 from 1.6.6 at the end of epoch 185, so there is always the chance that the different software version is having an impact.
